tools/omero/omero_get_id.py

```diff
@@ -0,0 +1,116 @@
+import argparse
+import csv
+import json
+import sys
+
+import ezomero as ez
+
+
+def get_ids_ezo(user, pws, host, port, final_obj_type, parent_obj_type, parent_id=None, tsv_file="id_list.tsv"):
+
+    # Function to write tabular file from the ezomero output
+    def write_ids_to_tsv(data):
+        with open(tsv_file, 'w', newline='') as f:
+            writer = csv.writer(f, delimiter='\t')
+            for item in data:
+                writer.writerow([item])  # Write each ID
+
+    with ez.connect(user, pws, "", host, port, secure=True) as conn:
+
+        if final_obj_type == "Project":
+            proj_ids = ez.get_project_ids(conn)
+            write_ids_to_tsv(proj_ids)
+            return proj_ids
+
+        elif final_obj_type == "Dataset":
+            args = {'project': None}
+            if parent_obj_type == "Project":
+                args['project'] = parent_id
+            ds_ids = ez.get_dataset_ids(conn, **args)
+            write_ids_to_tsv(ds_ids)
+            return ds_ids
+
+        elif final_obj_type == "Image":
+            args = {
+                'project': None,
+                'dataset': None,
+                'plate': None,
+                'well': None
```

Comment:
Are `well` and `plate` something specific to our setup or is this general for all OMERO servers?

Reply:
This is something general to OMERO servers.
You can either tidy your images in Image - Dataset - Project or Image - Well - Plate - Screen.
This is a project: https://idr.openmicroscopy.org/webclient/?show=project-2751
This is a screen: https://idr.openmicroscopy.org/webclient/?show=screen-3302

Reply:
I would say that for the moment ezomero supports well the first architecture and only partially the second, this is why I use omero-py which is closer to API and supports all but as soon as you don't have too many users on the second architecture, it is fine.